### PR TITLE
Remove Images.jl dep

### LINKED ISSUE #69: Remove Images.jl from ImageSegmentation.jl as dependency
- Would need to include ImageCore, ImageMorphology instead
- We need to resolve 1 function that only Images.jl offers
-> accum function
https://github.com/JuliaImages/Images.jl/blob/91a35ed4338c67e2f826b17a461eae612a026543/src/algorithms.jl#L253

### PR DESCRIPTION
**Pull request by ashwanirathee**
- [x] Removes Images.jl as dependencies but introduce ImageCore.jl + ImageMorphology.jl, reluctantly I had to add colorvectorspace which I thought won't be required as ImageCore.jl already requires it.

Still need to figure out what to do with `accum` function..

### CODE REVIEW

**Review by johnnychen94 — COMMENTED** [review]
Thanks! Can you also replace the test dependency "Images" with something smaller?

I noticed that you've limited CVS to 0.8, which is okay since it's broken for CVSv0.9 anyway, but we still need to bump to CVS 0.9. Are you going to upgrade ColorVectorSpace to v0.9 in this PR or for the next PR?

File: Project.toml [hunk 1]
```diff
@@ -4,9 +4,12 @@ version = "1.5.1"
 
 [deps]
 Clustering = "aaaa29a8-35af-508c-8bc3-b662a17a0fe5"
+ColorVectorSpace = "c3611d14-8923-5661-9e6a-0046d554d3a4"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 Distances = "b4f34e82-e78d-54a5-968a-f98e89d6e8f7"
+ImageCore = "a09fc81d-aa75-5fe9-8630-4744c3626534"
```
Comment (johnnychen94) [inline Project.toml:10]: Every entry in `deps` section needs an entry in the `compat` section.

File: src/ImageSegmentation.jl [hunk 2]
```diff
@@ -3,9 +3,14 @@ module ImageSegmentation
 import Base: show
 
 using LinearAlgebra, Statistics
-using Images, DataStructures, StaticArrays, ImageFiltering, LightGraphs, SimpleWeightedGraphs, RegionTrees, Distances, StaticArrays, Clustering, MetaGraphs
+using DataStructures, StaticArrays, ImageCore, ImageFiltering, ImageMorphology, LightGraphs, SimpleWeightedGraphs, RegionTrees, Distances, StaticArrays, Clustering, MetaGraphs
 import Clustering: kmeans, fuzzy_cmeans
 
+accum(::Type{T}) where {T<:Integer} = Int
```
Comment (johnnychen94) [inline src/ImageSegmentation.jl]: This seems like a good chance to rename it to `accum_type`.

Comment (ashwanirathee) [reply src/ImageSegmentation.jl]: I should keep it i core.jl right?

Comment (johnnychen94) [reply src/ImageSegmentation.jl]: I believe It's okay to just keep it here.

**Comment by ashwanirathee** [comment]
Actuallly Images.jl  provide two of its own functions(`findlocalmaxima` and `magnitude_phase`)  which aren't availble anywhere else but are used by this pkg's tests. So I'm not sure what I should do with that issue..I think we can upgrade this package to be comapatible with CVS 0.9 in next PR

File: Project.toml [hunk 3]
```diff
@@ -18,11 +20,14 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 
 [compat]
 Clustering = "0.10, 0.11, 0.12, 0.13, 0.14"
+ColorVectorSpace = "0.7, 0.8"
 DataStructures = "0.17.11, 0.18"
 Distances = "0.8, 0.9.2, 0.10"
 Documenter = "0.24, 0.25"
-ImageFiltering = "0.6"
 Images = "0.18, 0.19, 0.20, 0.21, 0.22, 0.23, 0.24"
```
Comment (johnnychen94) [inline Project.toml]:
```suggestion
```

**Review by johnnychen94 — APPROVED** [review]
LGTM

**Comment by johnnychen94** [comment]
Nice work. The next step is to add back ImageCore 0.9 and ColorVectorSpace 0.9 compatibility.

**Pull request merged**